frontend/src/upload_file.ts

```diff
@@ -155,7 +155,33 @@ class UploadFile {
     files.forEach(file => {
       const { fieldName, formId, renderer, uploads, uploadUrl } = this;
       const filename = file.name;
-      const uploadIndex = uploads.length;
+      let uploadIndex = uploads.length;
+
+      // #323 remove existing file
+      for (let index = 0; index < this.uploads.length; ++index) {
+        const existingUpload = this.uploads[index];
+        if (existingUpload instanceof Upload) {
+          if ((existingUpload.options as any).metadata.filename === filename) {
```

Comment:
I get a lint error on this line. This worked for me: `if (existingUpload.options?.metadata?.filename === filename) {`

Reply:
Is there a reason that the `metadata.filename` is used instead of `file.name`? Just curious, I don't have an opinion on this one.

Reply:
Because meta data is the original information that should exist and never been changed, and file.name could potentially be different (random bit etc although not used here). 

Away from my desk and cannot change the code, but I believe as maintainer you are allowed to push to our branch.

Reply:
It seems that I do have push access to your branch, to my surprise. I fixed the eslint warning. I also compiled `file_form.debug.js`.